### ISSUE TITLE
subscribeToMore: GraphQL subscription error (cannot read property 'then' of undefined)

### ISSUE THREAD

**Issue by muhammad-abubakkar** [apollographql/subscriptions-transport-ws#277]
apollo-client: 1.9.2
apollo-angular: 0.13.1
subscriptions-transport-ws: 0.8.3
angular: 4.4.3

**Comment by groundmuffin** [comment]
try to stay on graphql-js ^0.10.0

**Comment by shtse8** [comment]
I have the same issue, any updates?

**Comment by osmanperviz** [comment]
I have the same issue, any updates?

**Comment by AkimaLunar** [comment]
I am getting `{type: "error", id: "1", payload: {message: "Cannot read property 'then' of undefined"}}` error on subscription. What's the best way to debug? Where to look?

**Comment by gHashTag** [comment]
I'm Installed "graphql": "^0.10.5" and worked for me

**Comment by ianmartorell** [comment]
This was fixed with #261

**Comment by jcteague** [comment]
Has the fixed been released?  I had this problem.  Had to downgrade to graphql 0.10.5

**Comment by ianmartorell** [comment]
Yeah, upgrading to the latest version resolved it for me at the time.

**Comment by jcteague** [comment]
Hmmm... I must have some version conflicts going on.
```
"apollo-cache-inmemory": "^1.1.9",
 "apollo-client": "^2.0.4",
 "apollo-link-http": "^1.5.2",
 "apollo-link-ws": "^1.0.7",
 "graphql": "^0.10.5",
 "graphql-subscriptions": "^0.5.8",
 "subscriptions-transport-ws": "^0.9.0",
```

with graphql 0.11 I was gettting this error message, moving to 0.10.5 seemed to have worked (still testing)

**Comment by ianmartorell** [comment]
That's not the latest version of `subscriptions-transport-ws` though! I'm not sure what version introduced the fix, but these are the ones I'm on right now, everything working perfectly:
```
    "apollo-cache-inmemory": "^1.1.5",
    "apollo-client": "^2.2.0",
    "apollo-link": "^1.0.7",
    "apollo-link-http": "^1.3.2",
    "apollo-link-ws": "^1.0.4",
    "apollo-utilities": "^1.0.4",
    "graphql": "^0.12.3",
    "subscriptions-transport-ws": "^0.9.5"
```
Also, you don't need `graphql-subscriptions` on the client.

**Comment by jcteague** [comment]
Yeah, I've been playing a delicate version number dance to get this
working.  I'll try this particular cocktail and see if it works.

**Comment by Kikketer** [comment]
This issue is nearly a year old, and I'm getting the following error:
```
index.js:2177 Unhandled GraphQL subscription error TypeError: Cannot read property 'subscription' of undefined
    at SubscriptionHandshakeLink../node_modules/aws-appsync/lib/link/subscription-handshake-link.js.SubscriptionHandshakeLink.request
```
Should I still be using version 0.10?  Right now I'm using version 11.7 and am somewhat tied to that version since the `aws-appsync@1.2.1` dependency is using that version.

**Comment by ianmartorell** [comment]
You might have an old version of `subscriptions-transport-ws`. This issue was caused by `graphql` expecting a Promise from version ^0.11.0 onwards, and `subscriptions-transport-ws` not returning one. It was fixed and merged last September in #261, so I would recommend upgrading both of the packages to the latest versions you can, and you should be good. Going back to 0.10 is probably not a good idea, as there's bound to have been countless other bug fixes and improvements since then.

**Comment by Kikketer** [comment]
Thanks @ianmartorell .  I discovered the issue I had, I wasn't matching the arguments in my query to what I had defined in my schema server side.  All my libraries are up to speed now and working, the only one I had to downgrade was graphql from 0.13.2 to 0.13.0.

```
  "dependencies": {
    "apollo-cache-inmemory": "^1.2.6",
    "apollo-client": "^2.3.7",
    "apollo-link-http": "^1.5.4",
    "aws-appsync": "^1.3.3",
    "aws-appsync-react": "^1.1.3",
    "graphql": "0.13.0",  << Might be able to be updated to 0.13.2
    "graphql-tag": "^2.9.2",
    "react": "^16.1.1",
    "react-apollo": "^2.1.9",
    "react-dom": "^16.1.1"
  }
```

Be sure to match your gql query with what's defined in the schema (in my js client):
```
export default gql(`
subscription {
  createdEvent {
    name,
    when,
    where,
    description
  }
}`)
```

And my schema (defined on the server side in aws appsync):
```
type Subscription {
	createdEvent: Event
		@aws_subscribe(mutations: ["createEvent"])
}
```

Could probably close this issue.
